controllers/dscinitialization/monitoring.go

```diff
@@ -432,18 +432,28 @@ func createMonitoringProxySecret(ctx context.Context, cli client.Client, name st
 	return nil
 }
 
-func (r *DSCInitializationReconciler) configureCommonMonitoring(dsciInit *dsci.DSCInitialization) error {
-	// configure segment.io
-	segmentPath := filepath.Join(deploy.DefaultManifestPath, "monitoring", "segment")
-	if err := deploy.DeployManifestsFromPath(
-		r.Client,
-		dsciInit,
-		segmentPath,
-		dsciInit.Spec.ApplicationsNamespace,
-		"segment-io",
-		dsciInit.Spec.Monitoring.ManagementState == operatorv1.Managed); err != nil {
-		r.Log.Error(err, "error to deploy manifests under "+segmentPath)
-		return err
+func (r *DSCInitializationReconciler) configureCommonMonitoring(ctx context.Context, dsciInit *dsci.DSCInitialization) error {
+	// create segment.io only when configmap does not exist in the cluster
+	segmentioConfigMap := &corev1.ConfigMap{}
+	if err := r.Client.Get(ctx, client.ObjectKey{
+		Namespace: dsciInit.Spec.ApplicationsNamespace,
+		Name:      "odh-segment-key-config",
+	}, segmentioConfigMap); err != nil {
+		if apierrs.IsNotFound(err) {
+			segmentPath := filepath.Join(deploy.DefaultManifestPath, "monitoring", "segment")
+			if err := deploy.DeployManifestsFromPath(
+				r.Client,
+				dsciInit,
+				segmentPath,
+				dsciInit.Spec.ApplicationsNamespace,
+				"segment-io",
+				dsciInit.Spec.Monitoring.ManagementState == operatorv1.Managed); err != nil {
+				r.Log.Error(err, "error to deploy manifests under "+segmentPath)
+				return err
+			}
+			r.Log.Error(err, "error to get configmap 'odh-segment-key-config'")
```

Comment:
No, it's under `if apierrs.IsNotFound(err)` branch
I personally prefer to use functions for such cases, it allows to use early returns and avoid extra indentation levels, a-la
```golang
func (r *DSCInitializationReconciler) configureSegmentIO(ctx context.Context, dsciInit *dsci.DSCInitialization) error {
	// create segment.io only when configmap does not exist in the cluster
	segmentioConfigMap := &corev1.ConfigMap{}
	err := r.Client.Get(ctx, client.ObjectKey{
		Namespace: dsciInit.Spec.ApplicationsNamespace,
		Name:      "odh-segment-key-config",
	}, segmentioConfigMap)

	if err == nil {
		return nil
	}

	if !apierrs.IsNotFound(err) {
		r.Log.Error(err, "error to get configmap 'odh-segment-key-config'")
		return err
	}

	segmentPath := filepath.Join(deploy.DefaultManifestPath, "monitoring", "segment")
	if err := deploy.DeployManifestsFromPath(
		r.Client,
		dsciInit,
		segmentPath,
		dsciInit.Spec.ApplicationsNamespace,
		"segment-io",
		dsciInit.Spec.Monitoring.ManagementState == operatorv1.Managed); err != nil {
		r.Log.Error(err, "error to deploy manifests under "+segmentPath)
		return err
	}

	return nil
}


func (r *DSCInitializationReconciler) configureCommonMonitoring(ctx context.Context, dsciInit *dsci.DSCInitialization) error {
	if err := r.configureSegmentIO(ctx, dsciInit); err != nil {
		return err
	}
```
But it's a matter of taste

Reply:
i have extract logic of segment-io into a dedicated function as you suggested.